File: html/java/task3-form/processdata.java
import java.util.*;
import java.io.*;
import javax.servlet;

class JavaForm 
{
  public static void main( String args[] )
  {
    System.out.println("Content-type: text/html\r\n\r\n");

    System.out.println("<!DOCTYPE html>" +
      "<html lang='en'>" +
      "<head>" +
	    "<meta charset='UTF-8'>" +
	    "<title>Java CGI Form</title>" +
      "</head>" +
      "<body>" +
	    "<h1>Java Process Data</h1><hr />");

    String username;
    String password;
    int magicnum;


    System.out.println ("</body>" + "</html>");
  }
}


/*
// Create instance of FieldStorage 
form = cgi.FieldStorage() 

// Get data from fields
username = form.getvalue('username')
password  = form.getvalue('password')
magicnum = form.getvalue('magicnum')


print "Content-type:text/html\r\n\r\n"
print """
<!DOCTYPE html>
<html lang='en'>
<head>
	<meta charset="UTF-8">
	<title>Python CGI Form</title>
</head>
<body>
	<h1>Python Process Data</h1><hr />
"""
for i in range(0,int(magicnum)):
	print "<h1>Hello %s with a password of %s!</h1>" % (username, password)

print"""
</body>
</html>
"""
*/

File: html/go/processdata.go
package main

import (
    "net/http/cgi"
    "net/http"
    "fmt"
    "strconv"
)

func main() {

    fmt.Printf("Content-Type: text/html\n\n")

    var req *http.Request
    req, _ = cgi.Request()

    if(req.Method == "GET"){
        var queryString = req.URL.Query();
        var magnum, _ = strconv.Atoi(queryString.Get("magicnum"))
        printBody(queryString.Get("username"), queryString.Get("password"), magnum)
    } else if (req.Method == "POST"){
        username := req.FormValue("username");
        password := req.FormValue("password");
        magnum := req.FormValue("magicnum");
        magnumint, _ :=strconv.Atoi(magnum)
        printBody(username, password, magnumint)
    }
}


func printBody(username string, password string, n int){
    var helloString = "";
    for i := 0; i < n; i++ {
		helloString += fmt.Sprintf(`<h1>Hello %s with a password of %s!</h1>`, username, password)
	}

	fmt.Printf(`<!doctype html>
	<html lang="en">
	<head>
	<title>Node CGI Process Data</title>
	<style>
	</style>
	</head>
	<body>
		<h1>Node Process Data</h1>
		%s
	</body>
	</html>`, helloString);
}
